src/core/mavsdk_impl.h

```diff
@@ -85,6 +96,17 @@ class MavsdkImpl {
     bool _is_single_system{false};
 
     std::atomic<bool> _should_exit = {false};
+
+    void system_thread();
+    void send_heartbeat();
+
+    std::thread* _system_thread{nullptr};
+
+    /** @brief Interval in milliseconds to send heartbeat. */
+    // TODO kalyan - make this configurable??
```

Comment:
My answer is usually: no, let's leave it until someone asks.

Reply:
And that could go in a smaller, separate PR. So I'd argue that it is not in the scope of this PR anyway.